hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/LruCachedBlockQueue.java

```diff
@@ -65,14 +65,15 @@ public LruCachedBlockQueue(long maxSize, long blockSize) {
    * @param cb block to try to add to the queue
    */
   public void add(LruCachedBlock cb) {
```

Comment:
Well, this is not thread-safe anyway, so even with your change it is still susceptible to overflows. Not your problem but just wanted to point out.

Reply:
I was wondering if we should make the method synchronized and heapSize a volatile variable? What are your thoughts?